systems/citybuilding.go

```diff
@@ -1,74 +1,127 @@
 package systems
 
 import (
-	"fmt"
+	"image/color"
 
 	"engo.io/ecs"
 	"engo.io/engo"
+	"engo.io/engo/common"
+)
+
+const (
+	cityZIndex = 100
+
+	CityWidth  float32 = 30
+	CityHeight float32 = 30
 )
 
 type MouseTracker struct {
 	ecs.BasicEntity
-	engo.MouseComponent
+	common.MouseComponent
 }
 
-type City struct {
-	ecs.BasicEntity
-	engo.RenderComponent
-	engo.SpaceComponent
+type CityCategory uint8
+
+const (
+	CategoryRed CityCategory = iota
+	CategoryGreen
+	CategoryBlue
+)
+
+func (c CityCategory) String() string {
+	switch c {
+	case CategoryRed:
+		return "Red"
+	case CategoryGreen:
+		return "Green"
+	case CategoryBlue:
+		return "Blue"
+	default:
+		panic("CityCategory not found for String() method")
```

Comment:
Because it's a programmer-error, one that should be fixed before publishing the game.